src/main/java/net/dv8tion/jda/api/interactions/CustomIdInteraction.java

```diff
@@ -0,0 +1,29 @@
+package net.dv8tion.jda.api.interactions;
+
+import net.dv8tion.jda.api.interactions.components.ComponentInteraction;
+import net.dv8tion.jda.api.interactions.modals.ModalInteraction;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Interactions which have a custom ID, namely {@link ComponentInteraction} and {@link  ModalInteraction}.
+ *
+ * <br>This id does not have to be numerical.
+ *
+ */
+public interface CustomIdInteraction
+{
+    /**
+     * The custom ID provided to the component or modal when it was originally created.
+     * <br>This value should be used to determine what action to take in regard to this interaction.
```

Comment:
I don't think this is a good addition to the docs here. Certainly "should" is not the word I would use for this, and it also isn't really what I would describe it as. I think users can figure out what they would need this for.

```suggestion
```